drivers/watchdog/wdt_dw.c

```diff
@@ -11,55 +11,77 @@
 #include <zephyr/logging/log.h>
 #include <zephyr/sys_clock.h>
 #include <zephyr/math/ilog2.h>
+#include <zephyr/drivers/reset.h>
+#include <zephyr/drivers/clock_control.h>
 
 #include "wdt_dw.h"
 #include "wdt_dw_common.h"
 
 LOG_MODULE_REGISTER(wdt_dw, CONFIG_WDT_LOG_LEVEL);
 
-#define WDT_IS_INST_IRQ_EN(inst)	DT_NODE_HAS_PROP(DT_DRV_INST(inst), interrupts)
-#define WDT_CHECK_INTERRUPT_USED(inst)	WDT_IS_INST_IRQ_EN(inst) ||
-#define WDT_DW_INTERRUPT_SUPPORT	DT_INST_FOREACH_STATUS_OKAY(WDT_CHECK_INTERRUPT_USED) 0
-
 /* Device run time data */
 struct dw_wdt_dev_data {
+	/* MMIO mapping information for watchdog register base address */
+	DEVICE_MMIO_RAM;
+	/* Clock frequency */
+	uint32_t clk_freq;
 	uint32_t config;
-#if WDT_DW_INTERRUPT_SUPPORT
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(interrupts)
 	wdt_callback_t callback;
 #endif
 };
 
 /* Device constant configuration parameters */
 struct dw_wdt_dev_cfg {
-	uint32_t base;
-	uint32_t clk_freq;
-#if WDT_DW_INTERRUPT_SUPPORT
+	DEVICE_MMIO_ROM;
+
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(clocks)
+	/* Clock controller dev instance */
+	const struct device *clk_dev;
+	/* Identifier for timer to get clock freq from clk manager */
+	clock_control_subsys_t clkid;
+#endif
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(interrupts)
 	void (*irq_config)(void);
 #endif
 	uint8_t reset_pulse_length;
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(resets)
+	/* Reset controller device configurations */
+	struct reset_dt_spec reset_spec;
+#endif
 };
 
 static int dw_wdt_setup(const struct device *dev, uint8_t options)
 {
-	const struct dw_wdt_dev_cfg *const dev_config = dev->config;
+	int ret = 0;
 	struct dw_wdt_dev_data *const dev_data = dev->data;
+	uintptr_t reg_base = DEVICE_MMIO_GET(dev);
```

Comment:
Unnecessary assignment. And please place the `ret` declaration at the end (more common in Zephyr).
```suggestion
	struct dw_wdt_dev_data *const dev_data = dev->data;
	uintptr_t reg_base = DEVICE_MMIO_GET(dev);
	int ret;
```

Reply:
removed the initial value assignment which was not needed